Azure-Samples/chat-with-your-data-solution-accelerator
# Add Pre-commit hook

## Purpose
This work is Dependent on #243 hitting main first

- Split the requirements out into separate files for development (pytest/flake/black/etc)
- Add a pre-commit hook to run flake and black

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
- Rebuild devcontainer
- Make a bad formatting change (extra space at the end of a line)
- Try and commit and warning should appear
